### DISCUSSION TITLE
Port conflict when hosting DNS-server

### DISCUSSION THREAD

**Discussion by orzen** [k3s-io/k3s#4058] [Q&A] [answered]
Disclaimer, I've recently started to look into k8s and k3s so I have limited knowledge about how to run a cluster.

My goal is to host a DNS server in my cluster. I imagine that I have to configure traefik to route DNS traffic(udp/53) since the default configuration only accepts web(tcp/80) and websecure(tcp/443). Traefik crashes when I configure a traefik entrypoint for UDP port 53 since it says that the port is occupied by a different service. I assume this other service is kube-dns(coredns).

Are my assumptions correct and what would be the preferred way to solve this? Would it be possible to configure the cluster to use a different port for kube-dns or do I have to use DNS forwarding to handle a different port for the DNS server that I want to host?

**Comment by brandond** [comment, marked as answer]
Traefik is an ingress controller, which means it supports Ingress resources to route HTTP or HTTPS requests. If you want to expose DNS, I would recommend simply exposing a Service directly.
